packages/pdf-viewer/pdfSource.ts

```diff
@@ -71,4 +70,36 @@ export class PdfData {
 			scale,
 		};
 	};
+
+	public getActivePageNo = (scaledSize: ScaledSize, pageGap: number, scrollTop: number): number => {
+		const pageHeight = scaledSize.height + pageGap;
+		const pageNo = Math.floor(scrollTop / pageHeight) + 1;
+		return Math.min(pageNo, this.pageCount);
+	};
+
+	public printPdf = () => {
+		const frame = document.createElement('iframe');
```

Comment:
Pass "document" to the PdfData (now PdfDocument) class and make it a private property. In general class code shouldn't access the DOM directly or any global variable, as that makes it more difficult to test the code.